### PR TITLE
[CSM-113] refactored REST api

### PR DESCRIPTION
**Pull request by ksaric**
Added default 'skip' and 'limit' options, added a small section on README for running the wallet, updated the daedalus-bridge REST api.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,6 +59,16 @@ into which the project was cloned and run `stack build`.
...
+Running the wallet
+------------------
+
+After the project is built, run it:
+
+- open *new terminal window*, run `tmux` in it (`sudo apt-get install tmux` or else according to your distro)
+- open *new terminal window*, run `export WALLET_TEST=1; ./scripts/launch.sh` in it
+
+After a while, the first terminal windows should be populated with 4 tabs, 3 out of which will have a node running in them.
+
```
Comment (akegalj) [inline README.md]: Thanks for the steps. We are lacking documentation and functions should be better documented together with instructions. If you run into some functions (and we have them) and you know how they are working, feel free to document that code.

File: daedalus/src/Daedalus/BackendApi.purs [hunk 2]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
-updateWallet :: forall eff. CAddress -> CWalletMeta -> Aff (ajax :: AJAX | eff) CWallet
-updateWallet addr = postRBody ["update_wallet", _address addr]
-
--- FIXME: use DELETE method
 deleteWallet :: forall eff. CAddress -> Aff (ajax :: AJAX | eff) Unit
-deleteWallet addr = postR ["delete_wallet", _address addr]
+deleteWallet addr = deleteR ["wallets", _address addr]
+
+importKey :: forall eff. String -> Aff (ajax :: AJAX | eff) CWallet
+importKey = postRBody ["wallets/keys"]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: replace `["a/b"]` with `["a", "b"]`. `makeRequest` is taking care of that.

File: daedalus/src/Daedalus/BackendApi.purs [hunk 3]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
 
+restoreWallet :: forall eff. CWalletInit -> Aff (ajax :: AJAX | eff) CWallet
+restoreWallet = postRBody ["wallets/restore"]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: see https://github.com/input-output-hk/cardano-sl/pull/78/files#r104671937

File: daedalus/src/Daedalus/BackendApi.purs [hunk 4]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
 
-nextUpdate :: forall eff. Aff (ajax :: AJAX | eff) CUpdateInfo
-nextUpdate = getR ["next_update"]
+updateTransaction :: forall eff. CAddress -> CTxId -> CTxMeta -> Aff (ajax :: AJAX | eff) Unit
+updateTransaction addr ctxId = postRBody ["txs/payments", _address addr, _ctxIdValue ctxId]
 
-applyUpdate :: forall eff. Aff (ajax :: AJAX | eff) Unit
-applyUpdate = postR ["apply_update"]
+getHistory :: forall eff. CAddress -> Int -> Int -> Aff (ajax :: AJAX | eff) (Tuple (Array CTx) Int)
+getHistory addr skip limit = getR ["txs/histories", _address addr, show skip, show limit]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: see https://github.com/input-output-hk/cardano-sl/pull/78/files#r104671937

File: daedalus/src/Daedalus/BackendApi.purs [hunk 5]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
 
-restoreWallet :: forall eff. CWalletInit -> Aff (ajax :: AJAX | eff) CWallet
-restoreWallet = postRBody ["restore_wallet"]
+sendExtended :: forall eff. CAddress -> CAddress -> Coin -> CCurrency -> String -> String -> Aff (ajax :: AJAX | eff) CTx
+sendExtended addrFrom addrTo amount curr title desc = postR ["txs/payments", _address addrFrom, _address addrTo, show $ _coin amount, showCCurrency curr, title, desc]
 
-nextUpdate :: forall eff. Aff (ajax :: AJAX | eff) CUpdateInfo
-nextUpdate = getR ["next_update"]
+updateTransaction :: forall eff. CAddress -> CTxId -> CTxMeta -> Aff (ajax :: AJAX | eff) Unit
+updateTransaction addr ctxId = postRBody ["txs/payments", _address addr, _ctxIdValue ctxId]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: see https://github.com/input-output-hk/cardano-sl/pull/78/files#r104671937

File: daedalus/src/Daedalus/BackendApi.purs [hunk 6]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
+updateProfile = postRBody ["profile"]
+--------------------------------------------------------------------------------
+-- TRANSACTIONS ----------------------------------------------------------------
+send :: forall eff. CAddress -> CAddress -> Coin -> Aff (ajax :: AJAX | eff) CTx
+send addrFrom addrTo amount = postR ["txs/payments", _address addrFrom, _address addrTo, show $ _coin amount]
 
-restoreWallet :: forall eff. CWalletInit -> Aff (ajax :: AJAX | eff) CWallet
-restoreWallet = postRBody ["restore_wallet"]
+sendExtended :: forall eff. CAddress -> CAddress -> Coin -> CCurrency -> String -> String -> Aff (ajax :: AJAX | eff) CTx
+sendExtended addrFrom addrTo amount curr title desc = postR ["txs/payments", _address addrFrom, _address addrTo, show $ _coin amount, showCCurrency curr, title, desc]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: see https://github.com/input-output-hk/cardano-sl/pull/78/files#r104671937

File: daedalus/src/Daedalus/BackendApi.purs [hunk 7]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
+getProfile = getR ["profile"]
 
-blockchainSlotDuration :: forall eff. Aff (ajax :: AJAX | eff) Int
-blockchainSlotDuration = getR ["slot_duration"]
+updateProfile :: forall eff. CProfile -> Aff (ajax :: AJAX | eff) CProfile
+updateProfile = postRBody ["profile"]
+--------------------------------------------------------------------------------
+-- TRANSACTIONS ----------------------------------------------------------------
+send :: forall eff. CAddress -> CAddress -> Coin -> Aff (ajax :: AJAX | eff) CTx
+send addrFrom addrTo amount = postR ["txs/payments", _address addrFrom, _address addrTo, show $ _coin amount]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: see https://github.com/input-output-hk/cardano-sl/pull/78/files#r104671937

File: daedalus/src/Daedalus/BackendApi.purs [hunk 8]
```diff
@@ -74,70 +74,89 @@ postRBody urlPath content = flip makeRequest urlPath $
...
 
-applyUpdate :: forall eff. Aff (ajax :: AJAX | eff) Unit
-applyUpdate = postR ["apply_update"]
+getHistory :: forall eff. CAddress -> Int -> Int -> Aff (ajax :: AJAX | eff) (Tuple (Array CTx) Int)
+getHistory addr skip limit = getR ["txs/histories", _address addr, show skip, show limit]
 
-systemVersion :: forall eff. Aff (ajax :: AJAX | eff) SoftwareVersion
-systemVersion = getR ["system_version"]
+searchHistory :: forall eff. CAddress -> String -> Int -> Int -> Aff (ajax :: AJAX | eff) (Tuple (Array CTx) Int)
+searchHistory addr search skip limit = getR ["txs/histories", _address addr, search, show skip, show limit]
```
Comment (akegalj) [inline daedalus/src/Daedalus/BackendApi.purs]: https://github.com/input-output-hk/cardano-sl/pull/78/files#r104671937 

Please make these corrections in this file so I don't cp.

File: daedalus/src/Daedalus/BackendApi.purs (hunk 2 above)
Comment (ksaric) [inline daedalus/src/Daedalus/BackendApi.purs]: Done. On all the endpoints.

**Pull request merged**